server/WFCORE-4291_restore_legacy_not_graceful_startup_mode.adoc

```diff
@@ -0,0 +1,68 @@
+= WFCORE-4291 - Restore legacy (not "graceful") startup mode
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Since Wildfly 11, the server starts up in suspended mode, thus rejecting all incoming requests, until the server has completed booting, at which point the state is set to `RUNNING` and requests are accepted. While this helps ensure that the system is in a stable state before allowing any work to happen, it does have a negative effect on some deployment scenarios reported by Wildfly users. 
+
+One example scenario is this: if Deployment A depends -- at startup -- on external configuration, fetched by an HTTP request to Deployment B, Deployment A will fail to start as the HTTP request for that configuration will be rejected by the starting server.
+
+This RFE proposes a new startup mode, `graceless`, that will allow -- for some subsystems, at least -- requests to be handled much earlier in the application server boot process. In standalone mode, the system administrator can choose to enable this startup mode for that server, and in domain mode, the server group can be configured similarly, applying the configuration choice to all servers in that domain.
+
+== Issue Metadata
+
+=== Issue
+* https://issues.jboss.org/browse/WFCORE-4291[WFCORE-4291]
+
+=== Related Issues
+* https://issues.redhat.com/browse/EAP7-1320[EAP7-1320]
+
+=== Dev Contacts
+* mailto:{email}[{author}]
+
+=== QE Contacts
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+* [X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+For standalone mode, this option will be exposed as a command line option only, with the addition of a `--graceless` option. Since the value is needed before standalone.xml is parsed, it must come from the command line. Given this lack of start mode persistence, system administrators managing standalone instances will need to update their system scripts (`/etc/init.d`, `sysvinit`, `systemd`, etc.)
```

Comment:
What will happen if user starts server with `--graceless --start-mode=suspend`? Will have the graceless or suspend option preference?

Reply:
I can it add it here, yes, but I have already added it to the community doc section the Wildfly full PR: https://github.com/wildfly/wildfly/pull/13907/files#diff-bc1b4131dbdb71b96066a0ff68e2b144bfb7d85cd317f5b289e9f724286a8b6bR259-R261

Reply:
Rather than duplicating that info, I have added some more text describing the reload/restart scenarios, as well as the `start-mode=suspend` scenario to the community docs in the PR linked above.

Reply:
Thank you @jasondlee, but documentation and analysis are very different. Analysis document "defines" the feature and should scope out the requirements. What happens in the case I described is in my opinion a requirement. Documentation describes implemented feature for the user.

Reply:
These details were added yesterday.